### ISSUE TITLE
Function should not return values of different data-types

### ISSUE THREAD

**Issue by sandlerben** [hack4impact/maps4all#103]
`get_avg_ratings` (added in [926faff91f375129a34de1fb6a548e6cfd6cb3da](https://github.com/hack4impact/maps4all/commit/926faff91f375129a34de1fb6a548e6cfd6cb3da)) returns floats **or** strings depending on the state of the application. This is a very bad pattern because (a) it could easily lead to user TypeErrors and (b) to avoid a type error the caller must understand how this function works, which defeats the purpose of abstracting complexity away into a function.

**Comment by anniemeng** [comment]
how come it would return a string since we cast it to float?

**Comment by sandlerben** [comment]
I thought it would evaluate to a string (where is the float cast?) but also it doesn't even seem to evaluate

```
>>> total_sum = 5
>>> ratings = [1, 2, 3, 4, 5, 6, 7, 8]
>>> '%.1f' % total_sum / len(ratings)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for /: 'str' and 'int'
```

**Comment by anniemeng** [comment]
```
def get_avg_ratings():
    ratings = Rating.query.filter_by(resource_id=self.id).all()
    if not ratings:
      return -1.0

    total_sum = float(sum(r for r.rating in ratings))
    return '%.1f' % total_sum / len(ratings)
```
this is the function rn, and `total_sum` gets cast to a float and `len()` by default returns an int so it will always be a number?

**Comment by sandlerben** [comment]
Try running it in an interpreter

**Comment by anniemeng** [comment]
ah okay we're missing a parenthesis around the operation - i'll make a fix
<img width="299" alt="screen shot 2016-12-10 at 8 37 52 pm" src="https://cloud.githubusercontent.com/assets/5404536/21077412/8f954a6c-bf18-11e6-830e-d9c8946a663d.png">

**Comment by anniemeng** [comment]
actually rani who's using this rn will have it in her PR

also @yoninachmany who uses this function too

**Comment by anniemeng** [comment]
fixed in #105

**Comment by sandlerben** [comment]
this is why we should write tests
 





On Sun, Dec 11, 2016 2:15 AM, Annie Meng notifications@github.com
wrote:
fixed in #105
